bandit/core/metrics.py

```diff
@@ -32,21 +33,36 @@ def begin(self, fname):
         """Begin a new metric block.
 
         This starts a new metric collection name "fname" and makes is active.
-
         :param fname: the metrics unique name, normally the file name.
         """
-        self.data[fname] = {'loc': 0, 'nosec': 0}
+        self.data[fname] = {'loc': 0, 'nosec': 0, 'nosec_by_test': 0,
+                            'failed_nosec_by_test': 0}
         self.current = self.data[fname]
 
     def note_nosec(self, num=1):
-        """Note a "nosec" commnet.
+        """Note a "nosec" comment.
 
         Increment the currently active metrics nosec count.
-
         :param num: number of nosecs seen, defaults to 1
         """
         self.current['nosec'] += num
 
+    def note_nosec_by_test(self, num=1):
+        """Note a "nosec BXXX, BYYY, ..." comment.
+
+        Increment the currently active metrics nosec_by_test count.
+        :param num: number of nosecs seen, defaults to 1
+        """
+        self.current['nosec_by_test'] += num
+
+    def note_failed_nosec_by_test(self, num=1):
+        """Note a test failed not caught when specific tests in comment used
```

Comment:
I don't understand what this is tracking or what this sentence means

Reply:
I can reword this to make more sense. I meant that we were noting a test failure for a nosec comment that had the individual tests, but the test that failed was not one of the tests that the comment said to ignore.